### PR TITLE
feat: add functions required by SwayFX

### PR DESCRIPTION
**Pull request by WillPower3309**

### CODE REVIEW

**Comment by WillPower3309** [comment]
Seems windows with CSD decorations are broken:


Also, shadow scaling needs to be fixed

**Comment by ErikReider** [comment]
> Seems windows with CSD decorations are broken:
> 
> 
> Also, shadow scaling needs to be fixed

Should be a swayfx issue if I'm not mistaken. We're not providing the correct geometry to the renderer (xdg geometry for the xdg toplevels)

File: render/fx_renderer/fx_pass.c [hunk 1]
```diff
@@ -480,7 +480,7 @@ void fx_render_pass_add_box_shadow(struct fx_gles_render_pass *pass,
 	struct wlr_box shadow_box = options->shadow_box;
 	assert(shadow_box.width > 0 && shadow_box.height > 0);
 	struct wlr_box surface_box = options->clip_box;
-	float blur_sigma = shadow_data->blur_sigma * options->scale;
+	float blur_sigma = shadow_data->blur_sigma;
```
Comment (WillPower3309) [inline render/fx_renderer/fx_pass.c]: seems weird to scale this here when the comp would have already needed to scale it for damage tracking purposes. Open to thoughts though, I know we scale elsewhere in the rendering functions here, but maybe we shouldn't.

**Review by ErikReider — APPROVED** [review]
LGTM! :)

**Pull request merged**